ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -551,30 +551,10 @@ void McapHandler::set_on_disk_full_callback(
 
 void McapHandler::open_file_nts_()
 {
-    int mcap_file_index = 0;
-
-    if (configuration_.mcap_output_settings.files_max_size.size() > 0)
-    {
-        mcap_file_index = mcap_file_id_ % configuration_.mcap_output_settings.files_max_size.size();
-    }
-
-    // Update the file id
-    mcap_file_id_++;
-
-    if (configuration_.mcap_output_settings.file_rotation && mcap_filenames_.count(mcap_file_index))
-    {
-        const auto ret = std::filesystem::remove(mcap_filenames_[mcap_file_index]);
-
-        if (!ret)
-        {
-            logWarning(
-                DDSRECORDER_MCAP_HANDLER,
-                "RESOURCE_LIMITS | Failed to remove file " << mcap_filenames_[mcap_file_index] << " on file rotation.");
-        }
-    }
+    check_and_free_space_();
```

Comment:
NIT: a line comment wouldn't hurt.

Reply:
I was thinking about that, but in the upcoming PR I'll have to move it back to `open_mcap_file` since I would need to check every time I open a new one, in case a previous instance of the DdsRecorder had already filled up the available space.

Reply:
Let's see in the next PR, but I don't think that would be necessary; in that case open_file I guess would throw an exception and file rotation should kick in, freeing space if possible.